test/unit/services/validators/test-field-validator.js

```diff
@@ -0,0 +1,204 @@
+const expect = require('chai').expect
+const FieldValidator = require('../../../../app/services/validators/field-validator')
+const ErrorHandler = require('../../../../app/services/validators/error-handler')
+
+describe('services/validators/field-validator', function () {
+  const FIELD_NAME = 'Name'
+  const VALID_ALPHA = 'data'
+  const EMPTY_DATA = ''
+
+  var errorHandler
+
+  beforeEach(function () {
+    errorHandler = ErrorHandler()
+  })
+
+  describe('isRequired', function () {
+    it('should return an error object if passed null', function () {
+      var errorHandler = ErrorHandler()
+      FieldValidator(null, FIELD_NAME, errorHandler).isRequired()
+      var errors = errorHandler.get()
+      expect(errors).to.have.property(FIELD_NAME)
+    })
+
+    it('should return an error object if passed undefined', function () {
+      var errorHandler = ErrorHandler()
+      FieldValidator(undefined, FIELD_NAME, errorHandler)
+        .isRequired()
+      var errors = errorHandler.get()
+      expect(errors).to.have.property(FIELD_NAME)
+    })
+
+    it('should throw error if data is an object', function () {
+      var errorHandler = ErrorHandler()
+      FieldValidator({}, FIELD_NAME, errorHandler)
+        .isRequired()
+      var errors = errorHandler.get()
+      expect(errors).to.equal(false)
```

Comment:
Is this checking that an error is thrown?

Reply:
Nope - just that handler returns `false` if the internal collection is empty.

Reply:
So is the test description correct?

Reply:
Ah, sorry my mistake. It's actually the isRequired Handler that's wrong! I've added a lodash check for empty objects to the check:
`_.isEmpty(this.data)`